sled-hardware/Cargo.toml

```diff
@@ -10,6 +10,7 @@ anyhow.workspace = true
 camino.workspace = true
 cfg-if.workspace = true
 futures.workspace = true
+gethostname = "0.4.3"
```

Comment:
I'm somewhat shocked that this isn't built in.  Would it be better to use to use [nix's implementation](https://docs.rs/nix/latest/nix/unistd/fn.gethostname.html) (since we're already using `nix` I think)?

Either way, I'd think we'd want to add this to the workspace-level deps.

Reply:
Good idea. I'll use nix's impl!

Reply:
Turns out we are not using nix. It is a transitive dep though.